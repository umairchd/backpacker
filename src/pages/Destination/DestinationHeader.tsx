import React, { FC } from "react";
import { useServerContext } from "../lib/ServerContextProvider";
import { DestinationHeaderFragment } from "@@/pages/Destination/queries/destination-page-queries.generated";
import ReadMoreSection from "../Product/ProductContent/components/ReadMoreSection";

const DestinationHeader: FC<
  DestinationHeaderFragment & {
    name: string;
  }
> = ({ name, header, subheader, teaser }) => {
  const { channel } = useServerContext();
  const { features } = channel ?? {};

  return (
    <div>
      {name && (
        <h1 className="mt-5 text-38px md:text-40px font-bold text-black leading-normal">
          <span className="text-primary font-bold">{name} </span>Things To Do
        </h1>
      )}
      {!!header && <h2 className="mt-6px text-22px font-normal text-black mb-4">{header}</h2>}
      {features?.includes("listing-page-content") && (subheader || teaser) && (
        <>
          {!!subheader && <p className="text-base text-black mb-4 leading-33px">{subheader}</p>}
          {!!teaser && (
            <ReadMoreSection
              content={teaser}
              count={20}
            />
          )}
        </>
      )}
    </div>
  );
};

export default DestinationHeader;
