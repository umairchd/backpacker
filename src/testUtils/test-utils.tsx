import React, { FC, ReactElement, Suspense } from "react";
import { render, RenderOptions } from "@testing-library/react";

import { ThemeProvider } from "next-themes";

import "@@/styles/globals.scss";
import ServerContextProvider, { IServerContext } from "@@/pages/lib/ServerContextProvider";

import StoreProvider from "@@/pages/Store/StoreProvider";

import { MockedProvider, MockedResponse } from "@apollo/client/testing";

import defaultContextMock from "./mocks/serverContext.json";
import { ClientInfoProvider } from "@@/app/lib/ClientInfoProvider";

jest.mock("next/router", () => require("next-router-mock"));

global.window.matchMedia = jest.fn().mockReturnValue({
  matches: true,
  addListener: jest.fn(),
  removeListener: jest.fn(),
});

window.HTMLElement.prototype.scrollIntoView = jest.fn();

interface AllTheProvidersProps {
  children: React.ReactNode;
  contextMock?: any;
  apolloMocks?: ReadonlyArray<MockedResponse>;
}

export const AllTheProviders: FC<AllTheProvidersProps> = ({
  children,
  contextMock = defaultContextMock,
  apolloMocks,
}) => {
  const context: IServerContext = contextMock;

  return (
    <MockedProvider mocks={apolloMocks}>
      <ServerContextProvider value={context as any}>
        <StoreProvider>
          <ThemeProvider>
            <ClientInfoProvider host={context.host}>
              <Suspense>{children}</Suspense>
            </ClientInfoProvider>
          </ThemeProvider>
        </StoreProvider>
      </ServerContextProvider>
    </MockedProvider>
  );
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "wrapper">) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
