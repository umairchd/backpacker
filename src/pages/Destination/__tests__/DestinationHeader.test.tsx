/**
 * @jest-environment jest-environment-jsdom
 */

import React from "react";
import { render, cleanup } from "@testing-library/react";

import DestinationHeader from "../DestinationHeader";
import destinationHeaderMock from "./mocks/destinationHeaderMock.json";
import { AllTheProviders } from "@@/testUtils/test-utils";

describe.skip("when destination header is displayed", () => {
  let methods;

  beforeAll(() => {
    methods = render(
      <AllTheProviders>
        <DestinationHeader {...destinationHeaderMock} />
      </AllTheProviders>
    );
  });

  afterAll(cleanup);

  it("correct data must be rendered", () => {
    const { asFragment } = methods;

    expect(asFragment()).toMatchSnapshot();
  });
});
