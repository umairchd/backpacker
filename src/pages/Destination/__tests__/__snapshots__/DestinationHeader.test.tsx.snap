// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when destination header is displayed correct data must be rendered 1`] = `
<DocumentFragment>
  <div
    class="theme"
  >
    <div
      class="destination-header container-fluid"
    >
      <div
        class="header-image"
      >
        <picture>
          <source
            media="(min-width: 2400px)"
          />
          <source
            height="900"
            media="(min-width: 1600px)"
            width="1600"
          />
          <source
            height="787.5"
            media="(min-width: 1400px)"
            width="1400"
          />
          <source
            height="641.25"
            media="(min-width: 1140px)"
            width="1140"
          />
          <source
            height="576"
            media="(min-width: 1024px)"
            width="1024"
          />
          <source
            height="432"
            media="(min-width: 768px)"
            width="768"
          />
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-wave img-fluid css-1g5tya8-MuiSkeleton-root"
            style="width: 100%; height: 100%;"
          />
        </picture>
      </div>
      <div
        class="container"
      >
        <h1>
          TEST
        </h1>
        <h2>
          Test Header
        </h2>
        <a
          class="view-best-deal btn btn-primary"
          href="#top-three-activities"
          role="button"
          tabindex="0"
          type="button"
        >
          View Best Deals
        </a>
      </div>
    </div>
    <div
      class="destination-info container"
    >
      <div
        class="subheader"
      >
        Test Subheader
      </div>
      <div>
        Test Teaser
      </div>
    </div>
  </div>
</DocumentFragment>
`;
